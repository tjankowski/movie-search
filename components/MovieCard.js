import styles from "styles/MovieCard.module.scss";
import React from "react";
import Image from "components/Image";

export default function MovieCard({ item }) {
  return (
    <article className={styles.movieCard}>
      <Image src={item.Poster} title={item.Title} />
      <div className={styles.movieCard__title}>
        <h2>{item.Title}</h2>
        <div>{item.Year}</div>
        <div>{item.Type}</div>
      </div>
    </article>
  );
}
