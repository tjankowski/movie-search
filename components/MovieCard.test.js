import React from "react";
import renderer from "react-test-renderer";
import MovieCard from "./MovieCard";

describe("MovieCard", () => {
  it("should render without item", () => {
    expect(() => renderer.create(<MovieCard />)).toThrow();
  });

  it("should render item", () => {
    const item = {
      Poster: "/image.png",
      Title: "2020",
      Year: 2020,
      Type: "movie",
    };
    const component = renderer.create(<MovieCard item={item} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
