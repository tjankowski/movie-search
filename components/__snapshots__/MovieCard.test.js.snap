// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should render item 1`] = `
<article
  className="movieCard"
>
  <div
    className="image "
  >
    <img
      className="image__blur"
      src="/image.png"
    />
    <img
      className="image__poster"
      src="/image.png"
    />
  </div>
  <div
    className="movieCard__title"
  >
    <h3>
      2020
    </h3>
    <div>
      2020
    </div>
    <div>
      movie
    </div>
  </div>
</article>
`;
